### PR TITLE
added ST7796 model

### PR DESCRIPTION
**Pull request by liebman**
Added a model for ST7796.

Previously I used the ST7789 model and set the framebuffer size via options, since thats not possible now, I created a model for it.  Its exactly ST7789 but with the framebuffer size changed.

### CODE REVIEW

File: mipidsi/src/models/st7796.rs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        dcs.write_command(SetPixelFormat::new(pf))?;
+        delay.delay_us(10_000);
+        dcs.write_command(EnterNormalMode)?;
+        delay.delay_us(10_000);
+        dcs.write_command(SetDisplayOn)?;
+
+        // DISPON requires some time otherwise we risk SPI data issues
+        delay.delay_us(120_000);
+
+        Ok(madctl)
```
Comment (rfuest) [inline mipidsi/src/models/st7796.rs:62]:
```suggestion
        ST7789.init(dcs, delay, options, rst)
```

File: mipidsi/src/models/st7796.rs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        DI: WriteOnlyDataCommand,
+        I: IntoIterator<Item = Self::ColorFormat>,
+    {
+        dcs.write_command(WriteMemoryStart)?;
+
+        let mut iter = colors.into_iter().map(Rgb565::into_storage);
+
+        let buf = DataFormat::U16BEIter(&mut iter);
+        dcs.di.send_data(buf)?;
+        Ok(())
```
Comment (rfuest) [inline mipidsi/src/models/st7796.rs:76]:
```suggestion
        ST7789.write_pixels(dcs, colors)
```

File: mipidsi/README.md [hunk 3]
```diff
@@ -41,6 +41,7 @@ Variants that require different screen sizes and window addressing offsets are n
 * ILI9486
 * ST7735
 * ST7789
+* ST7796
```
Comment (rfuest) [inline mipidsi/README.md:44]: There is another copy of this list in `lib.rs`, which should also be updated.

**Review by rfuest — CHANGES_REQUESTED** [review]
If this model uses the same code as the ST7789 we could just reuse that implementation instead of duplicating the code.

You should also update the changelog to include the added model.

**Comment by liebman** [comment]
Updated with your suggested changes.  Also added other missing models from `lib.rs`.

**Review by rfuest — APPROVED** [review]
LGTM and good catch with the out of sync list in `lib.rs`.

**Pull request merged**